pkg/gpu/nvidia/health_check/health_checker.go

```diff
@@ -176,7 +179,14 @@ func (hc *GPUHealthChecker) listenToEvents() error {
 
 		// Only marking device unhealthy on Double Bit ECC Error
 		// See https://docs.nvidia.com/deploy/xid-errors/index.html#topic_4
-		if e.Edata != 48 {
+		xidIncluded := false
+		for _, xid := range hc.healthCriticalXid {
+			if e.Edata == xid {
+				xidIncluded = true
+				break
+			}
+		}
+		if !xidIncluded && e.Edata != 48 {
```

Comment:
We could add xid 48 to the list `healthCriticalXid`, and simplify this logic (i.e. remove special treatment for Xid 48).

Reply:
Done